src/vanilla/store.ts

```diff
@@ -165,16 +165,23 @@ type Pending = readonly [
   dependents: Map<AnyAtom, Set<AnyAtom>>,
   atomStates: Map<AnyAtom, AtomState>,
   functions: Set<() => void>,
+  finalizers: Set<AnyAtom>,
```

Comment:
I am hoping to use `functions` Set.

Reply:
Do we want finalization to happen after onMount is called?

Reply:
> Do we want finalization to happen after onMount is called?

Good question. That's what I thought I should ask. onMount is basically like useEffect, which can be async. So, syncEffect should be done in advance to onMount.

(So, I even wonder if "after flushPending" is the right approach.)

Reply:
Good point. React doesn't differentiate between useEffect and componentDidMount. Even the order of effects will affect output.

Reply:
Maybe we can just design store api apart from React lifecycle.
I'm fine to make `functions` an ordered list.